forecasting/src/autogluon/forecasting/task/forecasting/predictor.py

```diff
@@ -0,0 +1,182 @@
+import copy
+import logging
+
+import pandas as pd
+
+from .dataset import TimeSeriesDataset
+from core.task.base import BasePredictor
+from core.utils import plot_performance_vs_trials, plot_summary_of_models, plot_tabular_models, verbosity2loglevel
+from ...learner import AbstractLearner as Learner  # TODO: Keep track of true type of learner for loading
+from ...trainer import AbstractTrainer  # TODO: Keep track of true type of trainer for loading
+from core.utils.utils import setup_outputdir
+
+__all__ = ['ForecastingPredictor']
+
+logger = logging.getLogger()  # return root logger
+
+
+class ForecastingPredictor(BasePredictor):
+
+    def __init__(self, learner):
+        """ Creates TabularPredictor object.
+            You should not construct a TabularPredictor yourself, it is only intended to be produced during fit().
+
+            Parameters
+            ----------
+            learner : `AbstractLearner` object
+                Object that implements the `AbstractLearner` APIs.
+
+            To access any learner method `func()` from this Predictor, use: `predictor._learner.func()`.
+            To access any trainer method `func()` from this `Predictor`, use: `predictor._trainer.func()`.
+        """
+        self._learner: Learner = learner  # Learner object
+        self._trainer: AbstractTrainer = self._learner.load_trainer()  # Trainer object
+        self.output_directory = self._learner.path
+        self.eval_metric = self._learner.eval_metric
+
+    def get_model_names(self):
+        """Returns the list of model names trained in this `predictor` object."""
+        return self._trainer.get_model_names_all()
+
+    def predict(self, data, model=None):
+        forecasts, tss = self._learner.predict(data, model=model)
+        return forecasts, tss
+
+    def evaluate(self, data, quantiles=None):
+        perf = self._learner.score(data, quantiles=quantiles)
+        return perf
+
+    def evaluate_predictions(self, forecasts, tss, quantiles=None):
+
+        return self._learner.evaluate(forecasts, tss, quantiles=quantiles)
+
+    @classmethod
+    def load(cls, output_directory, verbosity=2):
+        if output_directory is None:
+            raise ValueError("output_directory cannot be None in load()")
+
+        output_directory = setup_outputdir(output_directory)  # replace ~ with absolute path if it exists
+        learner = Learner.load(output_directory)
+
+        return cls(learner=learner)
+
+    def save(self, output_directory):
+        """ Save this predictor to file in directory specified by this Predictor's `output_directory`.
+            Note that `fit()` already saves the predictor object automatically
+            (we do not recommend modifying the Predictor object yourself as it tracks many trained models).
+        """
+        self._learner.save()
+
+    def predict_proba(self, X):
+        pass
```

Comment:
can raise NotImplementedError for now

Reply:
I'm not sure if there is predict_proba for time series models?